docker/app.setup.sh

```diff
@@ -36,15 +38,31 @@ apt-get -y update
 apt-get -y install jq
 
 if [ ! -f "${PLUGINS_DIR}/wp-graphql/wp-graphql.php" ]; then
-    # WPGRAPHQL_VERSION in format like v1.2.3 or latest
-    echo "Install wp-graphql version (${WPGRAPHQL_VERSION})"
-    if [[ -z ${WPGRAPHQL_VERSION} || "${WPGRAPHQL_VERSION}" == "latest" ]]; then
-        echo "Installing latest WPGraphQL from WordPress.org"
-        wp plugin install wp-graphql --activate --allow-root
-    else
-    	echo "Installing WPGraphQL from Github"
-        wp plugin install "https://downloads.wordpress.org/plugin/wp-graphql.${WPGRAPHQL_VERSION-1.4.3}.zip" --allow-root --activate
-    fi
+
+
+	# if WPGRAPHQL_GIT_REPO is set, we'll install from the repo
+	if [[ -n ${WPGRAPHQL_GIT_REPO} ]]; then
+		echo "Installing WPGraphQL from GitHub repo ${WPGRAPHQL_GIT_REPO}"
+		# Clone the repository
+		git clone -b ${WPGRAPHQL_GIT_BRANCH} ${WPGRAPHQL_GIT_REPO} "${PLUGINS_DIR}/wp-graphql"
+		# Navigate to the plugin directory
+		cd "${PLUGINS_DIR}/wp-graphql"
+		# Install dependencies with Composer
+		composer install --no-dev
+		# Optionally activate the plugin using wp-cli
+		wp plugin activate wp-graphql --allow-root
```

Comment:
NOTE: 

This allows us to run tests against arbitrary branches of WPGraphQL. 

For example, this feature relies on some work in this PR (https://github.com/wp-graphql/wp-graphql/pull/3100)

So, I was able to update my `.env.testing` file with these values: 

```
 WPGRAPHQL_GIT_REPO="https://github.com/jasonbahl/wp-graphql.git"
 WPGRAPHQL_GIT_BRANCH="fix/interface-recursion"
```

And tests pass when running against that branch of WPGraphQL 👏🏻 👏🏻 

![CleanShot 2024-04-19 at 11 40 22](https://github.com/wp-graphql/wpgraphql-acf/assets/1260765/5bcda03b-0c5e-4a4e-8ca0-3de381aac66e)